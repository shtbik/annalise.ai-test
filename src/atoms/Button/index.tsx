import { FC, ButtonHTMLAttributes } from 'react'
import { StyledComponent } from '../../types/component'

import { StyledButton } from './styled.index'

interface IButton extends StyledComponent {}

const Button: FC<IButton & ButtonHTMLAttributes<HTMLButtonElement>> = ({ children, ...rest }) => {
  return <StyledButton {...rest}>{children}</StyledButton>
}

export default Button
