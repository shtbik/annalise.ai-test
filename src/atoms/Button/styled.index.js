import styled from 'styled-components'

export const StyledButton = styled.button`
  padding: 5px;

  &[aria-disabled='true'] {
    cursor: not-allowed;
  }

  &[aria-disabled='true']:active {
    pointer-events: none;
  }
`
