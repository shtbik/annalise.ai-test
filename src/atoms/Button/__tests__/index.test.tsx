import { render, fireEvent } from '@testing-library/react'
import styled from 'styled-components'

import Button from '../'

const SButton = styled(Button)`
  display: inline-block;
`

describe('Button component', () => {
  it('pass classname', () => {
    const { container } = render(<SButton>Click</SButton>)
    expect(container.firstChild).toHaveClass(SButton.styledComponentId)
    expect(container.firstChild).toHaveStyleRule('display', 'inline-block')
  })

  it('onClick', () => {
    const handleSubmit = jest.fn()
    const { container } = render(<Button onClick={handleSubmit}>Click</Button>)
    fireEvent.click(container.firstChild as ChildNode)
    fireEvent.click(container.firstChild as ChildNode)
    expect(handleSubmit).toBeCalledTimes(2)
  })

  it('snapshot', () => {
    const { container } = render(
      <Button type="submit" name="submit" disabled>
        Submit
      </Button>
    )
    expect(container).toMatchSnapshot()
  })
})
