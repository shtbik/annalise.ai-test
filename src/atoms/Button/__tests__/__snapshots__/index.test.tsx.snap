// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot 1`] = `
.c0 {
  padding: 5px;
}

.c0[aria-disabled='true'] {
  cursor: not-allowed;
}

.c0[aria-disabled='true']:active {
  pointer-events: none;
}

<div>
  <button
    class="c0"
    disabled=""
    name="submit"
    type="submit"
  >
    Submit
  </button>
</div>
`;
